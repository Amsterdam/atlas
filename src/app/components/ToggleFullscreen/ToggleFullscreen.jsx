import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button';
import Icon from '../Icon/Icon';

const ToggleFullscreen = ({ isFullscreen, title, onToggleFullscreen, alignLeft }) => (
  <Button
    title={isFullscreen ? `${title} verkleinen` : `${title} vergroten`}
    onClick={onToggleFullscreen}
    alignLeft={alignLeft}
  >
    <Icon icon={isFullscreen ? 'minimize' : 'maximize'} />
  </Button>
);

ToggleFullscreen.defaultProps = {
  alignLeft: false
};

ToggleFullscreen.propTypes = {
  alignLeft: PropTypes.bool,
  isFullscreen: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  onToggleFullscreen: PropTypes.func.isRequired
};

export default ToggleFullscreen;
