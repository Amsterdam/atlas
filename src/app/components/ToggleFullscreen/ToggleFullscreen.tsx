import { FunctionComponent } from 'react'
import styled, { css } from 'styled-components'
import { ControlButton } from '@amsterdam/arm-core'
import { themeSpacing } from '@amsterdam/asc-ui'
import Maximize from '../../../shared/assets/icons/icon-maximize.svg'
import Minimize from '../../../shared/assets/icons/icon-minimize.svg'

export interface ToggleFullscreenProps {
  alignLeft?: boolean
  isFullscreen: boolean
  title: string
  onToggleFullscreen: () => void
}

const StyledControlButton = styled(ControlButton)`
  position: absolute;

  top: ${themeSpacing(2)};
  z-index: 1;
  ${({ alignLeft }) =>
    alignLeft
      ? css`
          left: ${themeSpacing(2)};
        `
      : css`
          right: ${themeSpacing(2)};
        `}
`

const ToggleFullscreen: FunctionComponent<ToggleFullscreenProps> = ({
  isFullscreen,
  title,
  onToggleFullscreen,
  alignLeft,
}) => (
  <StyledControlButton
    variant="blank"
    title={isFullscreen ? `${title} verkleinen` : `${title} vergroten`}
    icon={isFullscreen ? <Minimize /> : <Maximize />}
    iconSize={28}
    size={40}
    onClick={onToggleFullscreen}
    alignLeft={alignLeft}
    extraClass="qa-toggle-fullscreen"
    className="qa-toggle-fullscreen"
  />
)

export default ToggleFullscreen
