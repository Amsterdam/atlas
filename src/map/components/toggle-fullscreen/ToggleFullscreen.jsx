import React from 'react';
import PropTypes from 'prop-types';

import MinimizeIcon from '../../../../public/images/icon-minimize.svg';
import MaximizeIcon from '../../../../public/images/icon-maximize.svg';

const ToggleFullscreen = ({ isFullscreen, onToggleFullscreen }) => (
  <button
    className="toggle-fullscreen"
    onClick={() => onToggleFullscreen(!isFullscreen)}
  >
    {isFullscreen ? <MinimizeIcon /> : <MaximizeIcon />}
  </button>
);

ToggleFullscreen.propTypes = {
  isFullscreen: PropTypes.bool.isRequired,
  onToggleFullscreen: PropTypes.func.isRequired
};

export default ToggleFullscreen;
