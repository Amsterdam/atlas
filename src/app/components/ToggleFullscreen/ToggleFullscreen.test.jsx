import React from 'react';
import { shallow } from 'enzyme';

import ToggleFullscreen from './ToggleFullscreen';

describe('ToggleFullscreen', () => {
  const onToggleFullscreen = jest.fn();
  const props = {
    isFullscreen: true,
    title: 'ABC',
    onToggleFullscreen
  };

  describe('actions', () => {
    it('should trigger toggle off when clicked', () => {
      const wrapper = shallow(
        <ToggleFullscreen {...props} />
      );
      wrapper.find('button').at(0).simulate('click');
      expect(onToggleFullscreen).toHaveBeenCalled();
    });

    it('should trigger toggle on when clicked', () => {
      const wrapper = shallow(
        <ToggleFullscreen {...props} />
      );
      wrapper.find('button').at(0).simulate('click');
      expect(onToggleFullscreen).toHaveBeenCalled();
    });

    describe('rendering', () => {
      it('should render with fullscreen is turned on', () => {
        const wrapper = shallow(
          <ToggleFullscreen {...props} />
        );
        expect(wrapper).toMatchSnapshot();
      });

      it('should render with fullscreen is turned off', () => {
        const wrapper = shallow(
          <ToggleFullscreen {...props} />
        );
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
