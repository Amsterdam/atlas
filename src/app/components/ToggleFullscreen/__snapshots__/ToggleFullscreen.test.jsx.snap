// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleFullscreen actions rendering should render with fullscreen is turned off 1`] = `
<Button
  alignLeft={false}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
        Array [],
      ],
    }
  }
  title="ABC verkleinen"
>
  <Icon
    alignLeft={false}
    icon="minimize"
  />
</Button>
`;

exports[`ToggleFullscreen actions rendering should render with fullscreen is turned on 1`] = `
<Button
  alignLeft={false}
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
        Array [],
      ],
    }
  }
  title="ABC verkleinen"
>
  <Icon
    alignLeft={false}
    icon="minimize"
  />
</Button>
`;
